docs/lang/articles/visualization/ggui.md

```diff
@@ -108,6 +109,212 @@ If a mesh has `num` triangles, the `indices` should be a 1D scalar field with a
 
 `normals` is an optional parameter for `scene.mesh()`.
 
+:::example
+
+1. Example of drawing 3d-lines
+
+```python
+import taichi as ti
+
+ti.init(arch=ti.cuda)
+
+N = 10
+
+particles_pos = ti.Vector.field(3, dtype=ti.f32, shape = N)
+points_pos = ti.Vector.field(3, dtype=ti.f32, shape = N)
+
+@ti.kernel
+def init_points_pos(points : ti.template()):
+    for i in range(points.shape[0]):
+        points[i] = [i for j in ti.static(range(3))]
+
+init_points_pos(particles_pos)
+init_points_pos(points_pos)
+
+window = ti.ui.Window("Test for Drawing 3d-lines", (768, 768))
+canvas = window.get_canvas()
+scene = ti.ui.Scene()
+camera = ti.ui.make_camera()
+camera.position(5, 2, 2)
+
+while window.running:
+    camera.track_user_inputs(window, movement_speed=0.03, hold_key=ti.ui.RMB)
+    scene.set_camera(camera)
+    scene.ambient_light((0.8, 0.8, 0.8))
+    scene.point_light(pos=(0.5, 1.5, 1.5), color=(1, 1, 1))
+
+    scene.particles(particles_pos, color = (0.68, 0.26, 0.19), radius = 0.1)
+    # Here draw 3d-lines in the scene
+    scene.lines(points_pos, color = (0.28, 0.68, 0.99), width = 5.0)
+    canvas.scene(scene)
+    window.show()
+```
+
+### Advanced 3d Geometries
+
+```python
+scene.lines(vertices, width, indices, color, per_vertex_color, vertex_offset, vertex_count, index_offset, index_count)
+
+scene.mesh(vertices, indices, normals, color, per_vertex_color, vertex_offset, vertex_count, index_offset, index_count, show_wareframe)
```

Comment:
w**i**reframe

Reply:
ok so this is not only a typo, the wrong spelling is actually in the code repo.. So before releasing this doc we might have to fix the API too. And please consider to use a shorter name like `wireframe`.

Reply:
ok I will fix this